### PR TITLE
Noy/authentication

### PR DESCRIPTION
**Pull request by NoyRobinson**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/employeeFeedback/dao/models.kt [hunk 1]
```diff
@@ -4,6 +4,14 @@ import java.sql.Date
 
 data class EmployeeInfo(val id: Int, val role: RoleType, val department: String, val companyId: Int)
 
+data class EmployeeCreation(val firstName: String, val lastName: String, val role: RoleType, val department: String, val companyId: Int)
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/dao/models.kt]: It's better to call it EmployeeCreationRequest, so it's clear that it holds data, not doing an action

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/AuthenticationService.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class AuthenticationService(private val employeeDao: EmployeeDao) {
+
+    companion object{
+        const val SECRET_KEY = "kgjhdkjgbnsdjgbejr43775384957934579fhdjbdjsnks"
+    }
+
+    fun authenticate(id: Int, firstName: String, lastName: String): LoggedInEmployeeInfo {
+        val loginParams = LoginParams(id, firstName, lastName)
+        val loggedInEmployeeInfo = employeeDao.findEmployeeByLoginParams(loginParams)
+            ?: throw Exception("Authentication failed")
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/service/AuthenticationService.kt]: Make sure that the user gets a "Not authorized" / "No access" exception, not an internal server error.
It's more user friendly

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 3]
```diff
@@ -21,24 +19,31 @@ class AuthenticationFilter: ContainerRequestFilter {
...
         cookie.let {
             try {
-                Jwts.parser().setSigningKey(SECRET_KEY)
+                val claims = Jwts.parser()
+                                    .setSigningKey(SECRET_KEY)
+                                    .parseClaimsJws(cookie)
+                                    .body
+                requestContext.setProperty("employeeId", claims["id"])
+                requestContext.setProperty("companyId", claims["companyId"])
+                requestContext.setProperty("companyId", claims["companyId"])
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: you have companyId twice here, did you mean to put something else?

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 4]
```diff
@@ -47,3 +52,5 @@ class AuthenticationFilter: ContainerRequestFilter {
         }
     }
 }
+
+
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: remove one blank line

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt (hunk 3 above)
Comment (NoyRobinson) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: Yes, my mistake. Fixed it

**Review by vasofi — APPROVED** [review]

**Pull request merged**